shard/shard.go

```diff
@@ -20,40 +20,78 @@
 
 package shard
 
+const (
+	// Initializing represents a shard newly assigned to an instance
+	Initializing State = iota
+
+	// Available represends a shard bootstraped and ready to serve
+	Available
+
+	// Leaving is laeving
```

Comment:
Typo here, also perhaps make it more similar to the other comments: "Leaving represents a shard that is intending to be removed"